### ISSUE TITLE
Cinnamon crashes after updating Nvidia driver to version 367.57

### ISSUE THREAD

**Issue by bararan** [linuxmint/Cinnamon#5899]
After updating my Nvidia driver from version to 367.57 this evening Cinnamon started to crash after login and I wasn't able to restart Cinnamon. After switching to Xorg driver from driver manager things went back to normal.

Not sure I should be reporting this here, so apologies if this is not the place.

**Comment by leigh123linux** [comment]
367.57 works fine here, without any logs this issue will never be looked at or resolved.
I you reinstall nvidia run this command and post the file

```
nvidia-bug-report.sh
```

If not you might as well close it.

**Comment by bararan** [comment]
Thanks for checking this.

I switched back to nvidia 367.57 and the problem repeated. I ran the bug report script. The log file is attached below. Hope it holds some clue to what might be causing the crash.
[nvidia-bug-report.log.gz](https://github.com/linuxmint/Cinnamon/files/570930/nvidia-bug-report.log.gz)

**Comment by pandreetto** [comment]
Hi all
I think I have the same problem but with nvidia-304-updates (304.132-0ubuntu0.16.04.2)
The report is in attach
[nvidia-bug-report.log.gz](https://github.com/linuxmint/Cinnamon/files/570953/nvidia-bug-report.log.gz)
Thanks

**Comment by leigh123linux** [comment]
@bararan 
So you have optimus

```
[    30.640] (--) PCI:*(0:0:2:0) 8086:0116:1043:1682 rev 9, Mem @ 0xdd400000/4194304, 0xb0000000/268435456, I/O @ 0x0000e000/64
[    30.640] (--) PCI: (0:1:0:0) 10de:1050:1043:1682 rev 161, Mem @ 0xdc000000/16777216, 0xc0000000/268435456, 0xd0000000/33554432, I/O @ 0x0000d000/128, BIOS @ 0x????????/524288
```
bbswitch has selected nvidia hardware ok
```
[   25.111254] bbswitch: version 0.8
[   25.111262] bbswitch: Found integrated VGA device 0000:00:02.0: \_SB_.PCI0.GFX0
[   25.111268] bbswitch: Found discrete VGA device 0000:01:00.0: \_SB_.PCI0.PEG0.GFX0
[   25.111278] ACPI Warning: \_SB_.PCI0.PEG0.GFX0._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20150930/nsarguments-95)
[   25.111395] bbswitch: detected an Optimus _DSM function
[   25.111405] bbswitch: Succesfully loaded. Discrete card 0000:01:00.0 is on
```

errors in log are

```
[    34.700] (EE) [drm] Failed to open DRM device for (null): -22
[    35.078] (EE) Failed to initialize GLX extension (Compatible NVIDIA X driver not found)
```

You seem to have no nvidia config files which is the cause of the issue IMO

```
____________________________________________

*** /usr/share/X11/xorg.conf.d/50-nvidia-drm-outputclass.conf: does not exist

____________________________________________

*** nvidia-drm-outputclass.conf does not exist

____________________________________________
```

**Comment by leigh123linux** [comment]
@pandreetto 

You issue is simple, downgrade to the older driver as 304.132 is known to be broken


http://askubuntu.com/questions/840338/16-10-and-nvidia-304-132https://devtalk.nvidia.com/default/topic/972001/linux/serious-regression-in-legacy-304xx-304-132-drivers/

https://bugzilla.rpmfusion.org/show_bug.cgi?id=4288

https://devtalk.nvidia.com/default/topic/968892/geforce-7300le-with-304-132-glxinfo-fails-now-with-quot-x-error-of-failed-request-badvalue-quot-/

https://devtalk.nvidia.com/default/topic/972001/linux/serious-regression-in-legacy-304xx-304-132-drivers/

**Comment by leigh123linux** [comment]
Neither of these issues are related to cinnamon and the report here should be closed.
You need to re-file the issues against nvidia instead, probably at launchpad bug tracking.

**Comment by pandreetto** [comment]
Many thanks for the suggestion and the time  spent.

**Comment by barisumog** [comment]
@bararan 

I had the same issue on acer laptop with gtx-850m optimus card. After yesterday's update, cinnamon would crash to fall back mode if using nvidia. This is how I managed to clear the issue:

1. update to the latest kernel (4.4.0-45)
2. recompile driver
`sudo dpkg-reconfigure nvidia-367`
3. reconfigure mdm
`sudo dpkg-reconfigure mdm`

After reboot, both nvidia and intel now works fine as before.

**Comment by bararan** [comment]
Thanks to everyone for contributions.

@leigh123linux it's weird that no nvidia log file could be detected. After reading your reply I checked the directory where it should be and found it there. But I agree that this appears to be an Nvidia problem, so I'll close the issue after this comment.

@barisumog thank you for the suggestion. Unfortunately it didn't do the trick for me. I'll stick with the Xorg drivers for the time being. The reason why I switched to Nvidia drivers in the first place was that with Xorg my cursor used to leave a trail, which was annoying. That doesn't seem to be the case anymore, so the Xorg drivers should work just fine for the time being.

**Comment by TassosD** [comment]
I had the same issue and after Cinnamon crash I could work with a destroyed desktop UI. In my laptop I have 2 graphic cards (Nvidia GF108M and a on-board Intel). When updating Nvidia driver I had selected the Intel card. I did the following:
1. opened NVIDIA X Sever Settings from start menu
2. chose Intel Card (Nvidia card was mistakenly selected)
3. restarted system and problem had been resolved.

**Comment by clefebvre** [comment]
Hi,

Can you try http://filebin.ca/31C3G2wMipLC/ubuntu-drivers-common_0.4.17linuxmint2_amd64.deb and see if it helps?

**Comment by clefebvre** [comment]
I'm testing on an ASUS laptop here with an NVIDIA GTX 850M and I can't reproduce the issue. Everything's working fine. 

It's a fresh install though, so it installed the 367 drivers directly... it never had the 361 before that. Maybe that's why.

Or maybe it's related to the use of bumblebee? Do you have bumblebee installed?

In any case, please try http://filebin.ca/31C3G2wMipLC/ubuntu-drivers-common_0.4.17linuxmint2_amd64.deb.

In https://bugs.launchpad.net/ubuntu/+source/ubuntu-drivers-common/+bug/1619306 I can see:

> The new driver also requires changes in ubuntu-drivers-common, as gpu-manager needs to be able to unload the nvidia-drm module in order to power down the dGPU in hybrid systems.

This new package adds support for that. So although I can't reproduce the problem and observe a solution, it certainly looks pertinent.

Make sure you reboot after installing this package.

**Comment by bararan** [comment]
@clefebvre, your solution did work. I've just reinstalled nvidia driver after installing the ubuntu drivers common package from the link you provided and cinnamon starts without any problems now. I also tried switching between Intel and Nvidia gpu's and no problems there either.

Thanks a lot!

**Comment by clefebvre** [comment]
Ok that's great news. We can start working on rolling out the fix.

I'd like to understand a little more before though...

Can you paste the content of the two following files for me?

/var/log/apt/history.log
/var/log/dpkg.log

Thanks.

**Comment by bararan** [comment]
Sure. This is the **/var/log/apt/history.log**:

```
Start-Date: 2016-11-01  17:44:19
Commandline: /usr/sbin/synaptic --hide-main-window --non-interactive --parent-window-id 41943076 -o Synaptic::closeZvt=true --progress-str Please wait, this can take some time --finish-str Update is complete --set-selections-file /tmp/tmpm14f19yv
Requested-By: bararan (1000)
Upgrade: libdns-export162:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), libisccfg140:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), bind9-host:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), dnsutils:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), libisc160:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), libisc-export160:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), liblwres141:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), oxideqt-codecs-extra:amd64 (1.17.9-0ubuntu0.16.04.1, 1.18.3-0ubuntu0.16.04.1), libdns162:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), libisccc140:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2), libbind9-140:amd64 (1:9.10.3.dfsg.P4-8ubuntu1.1, 1:9.10.3.dfsg.P4-8ubuntu1.2)
End-Date: 2016-11-01  17:44:40

Start-Date: 2016-11-02  14:58:43
Requested-By: bararan (1000)
Install: git-man:amd64 (1:2.7.4-0ubuntu1, automatic), git:amd64 (1:2.7.4-0ubuntu1), liberror-perl:amd64 (0.17-1.2, automatic)
End-Date: 2016-11-02  14:58:55

Start-Date: 2016-11-02  17:05:10
Commandline: apt-get install vim
Requested-By: bararan (1000)
Install: vim-runtime:amd64 (2:7.4.1689-3ubuntu1.1, automatic), vim:amd64 (2:7.4.1689-3ubuntu1.1)
End-Date: 2016-11-02  17:05:26

Start-Date: 2016-11-03  11:23:25
Commandline: /usr/sbin/synaptic --hide-main-window --non-interactive --parent-window-id 48234528 -o Synaptic::closeZvt=true --progress-str Please wait, this can take some time --finish-str Update is complete --set-selections-file /tmp/tmpo6ycs24u
Requested-By: bararan (1000)
Upgrade: openjdk-8-jre:amd64 (8u91-b14-3ubuntu1~16.04.1, 8u111-b14-2ubuntu0.16.04.2), openjdk-8-jre-headless:amd64 (8u91-b14-3ubuntu1~16.04.1, 8u111-b14-2ubuntu0.16.04.2)
End-Date: 2016-11-03  11:23:42

Start-Date: 2016-11-03  21:44:28
Commandline: /usr/sbin/synaptic --hide-main-window --non-interactive --parent-window-id 48234528 -o Synaptic::closeZvt=true --progress-str Please wait, this can take some time --finish-str Update is complete --set-selections-file /tmp/tmp35y58pgw
Requested-By: bararan (1000)
Install: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1, automatic), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1, automatic), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1, automatic)
Upgrade: libcurl3:amd64 (7.47.0-1ubuntu2.1, 7.47.0-1ubuntu2.2), libcuda1-361:amd64 (361.42-0ubuntu2, 367.57-0ubuntu0.16.04.1), curl:amd64 (7.47.0-1ubuntu2.1, 7.47.0-1ubuntu2.2), nvidia-opencl-icd-361:amd64 (361.42-0ubuntu2, 367.57-0ubuntu0.16.04.1), libcurl3-gnutls:amd64 (7.47.0-1ubuntu2.1, 7.47.0-1ubuntu2.2), nvidia-361:amd64 (361.42-0ubuntu2, 367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-03  21:48:05

Start-Date: 2016-11-04  01:32:17
Commandline: aptdaemon role='role-commit-packages' sender=':1.49'
Install: intel-microcode:amd64 (3.20151106.1), iucode-tool:amd64 (1.5.1-1, automatic)
End-Date: 2016-11-04  01:32:53

Start-Date: 2016-11-04  01:37:33
Commandline: aptdaemon role='role-commit-packages' sender=':1.48'
Install: libcuda1-304:amd64 (304.132-0ubuntu0.16.04.2), xserver-xorg-legacy:amd64 (2:1.18.4-0ubuntu0.1, automatic), nvidia-opencl-icd-304:amd64 (304.132-0ubuntu0.16.04.2), nvidia-304:amd64 (304.132-0ubuntu0.16.04.2)
Upgrade: xserver-common:amd64 (2:1.18.3-1ubuntu2, 2:1.18.4-0ubuntu0.1)
Remove: libcuda1-361:amd64 (367.57-0ubuntu0.16.04.1), libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-361:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-361:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-04  01:39:43

Start-Date: 2016-11-04  01:42:35
Commandline: aptdaemon role='role-commit-packages' sender=':1.47'
Remove: nvidia-304:amd64 (304.132-0ubuntu0.16.04.2)
End-Date: 2016-11-04  01:43:27

Start-Date: 2016-11-04  04:08:08
Commandline: aptdaemon role='role-commit-packages' sender=':1.132'
Install: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
Remove: libcuda1-304:amd64 (304.132-0ubuntu0.16.04.2), nvidia-opencl-icd-304:amd64 (304.132-0ubuntu0.16.04.2)
End-Date: 2016-11-04  04:11:29

Start-Date: 2016-11-04  04:29:09
Commandline: aptdaemon role='role-commit-packages' sender=':1.58'
Remove: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-04  04:30:03

Start-Date: 2016-11-04  10:25:28
Commandline: aptdaemon role='role-commit-packages' sender=':1.53'
Install: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-04  10:29:01

Start-Date: 2016-11-04  10:52:44
Commandline: aptdaemon role='role-commit-packages' sender=':1.52'
Remove: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-04  10:53:38

Start-Date: 2016-11-07  17:41:30
Commandline: aptdaemon role='role-commit-packages' sender=':1.43'
Install: libcuda1-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-opencl-icd-367:amd64 (367.57-0ubuntu0.16.04.1), nvidia-367:amd64 (367.57-0ubuntu0.16.04.1)
End-Date: 2016-11-07  17:45:01```

**Comment by bararan** [comment]
I've uploaded **/var/log/dpkg.log** as a file (converted to a txt file; apparently log isn't a supported file format) as it's rather long: 
[dpkglog.txt](https://github.com/linuxmint/Cinnamon/files/577105/dpkglog.txt)

**Comment by clefebvre** [comment]
Thanks

**Comment by Fhrozen** [comment]
Good Evening, I am new using Mint. Try to migrate from ubuntu.
Now, the problem is i am a DNN user, so i need to use cuda (NVIDIA). I tried to install cuda 8.0 to mint 18.1 cinnamon using the deb from ubuntu 16.04 of NVIDIA. I have no problem of running cuda but i cinnamon desktop is only running on fallback mode.
My PC is a desktop intel board MSI and a video nvidia titan. I tried to configure it by generating a xorg.conf on command mode, but when i move it to /etc/X11 and restart it, it even gets worse.
I am not sure if this is a different issue or is related with this one. The drivers installed by cuda 8.0 are v367.57
